opentelemetry-sdk/src/opentelemetry/sdk/_logs/_internal/__init__.py

```diff
@@ -618,6 +617,7 @@ def __init__(
     def resource(self):
         return self._resource
 
+    @lru_cache(maxsize=None)
```

Comment:
Is the point of this only to speed up the multiple calls to get logger? Or is there a functionality benefit. Given that the logs will be coming from multiple loggers, I am not sure this cache is necessary.

Reply:
Yes, this is only to speed up the multiple calls to get_logger. We can be more explicit in keeping track of one logger instance per logger name encountered. `@lru_cache` may not be the right approach here if we want to be able to retrieve all logger instances after they've been placed in the cache.

Reply:
"@lru_cache may not be the right approach here if we want to be able to retrieve all logger instances after they've been placed in the cache"
Not sure I understood this. Could you explain further?

Please add cache tests and the pytest benchmarking you mentioned. Otherwise, looks great to me!